includes/abstracts/abstract-wc-product.php

```diff
@@ -1493,7 +1503,7 @@ public function has_weight() {
 	 * @return bool
 	 */
 	public function is_in_stock() {
```

Comment:
We should add inline documentation to indicate that this returns true for onbackorder items (bw compat).

Reply:
That said, back order items are not in stock, but can be ordered. I know this is a BW compatibility issue and probably cannot be corrected?

Reply:
`is_in_stock` previously returned true for items that allow backorders and have negative inventory, so the behavior is not really changing (I'll still add inline docs for it). You're correct that we have to keep this behavior for backwards compatibility.